tesseract/tesseract_environment/include/tesseract_environment/core/types.h

```diff
@@ -0,0 +1,243 @@
+/**
+ * @file types.h
+ * @brief The tesseract_environment package types.
+ *
+ * @author Levi Armstrong
+ * @date April 15, 2018
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2013, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_ENVIRONMENT_TYPES_H
+#define TESSERACT_ENVIRONMENT_TYPES_H
+
+#include <tesseract_environment/core/macros.h>
+TESSERACT_ENVIRONMENT_IGNORE_WARNINGS_PUSH
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <Eigen/StdVector>
+#include <unordered_map>
+#include <vector>
+#include <memory>
+#include <functional>
+#include <map>
+#include <tesseract_scene_graph/graph.h>
+TESSERACT_ENVIRONMENT_IGNORE_WARNINGS_POP
+
+namespace tesseract_environment
+{
+template <typename T>
+using AlignedVector = std::vector<T, Eigen::aligned_allocator<T>>;
+
+template <typename Key, typename Value>
+using AlignedMap = std::map<Key, Value, std::less<Key>, Eigen::aligned_allocator<std::pair<const Key, Value>>>;
+
+template <typename Key, typename Value>
+using AlignedUnorderedMap = std::unordered_map<Key,
+                                               Value,
+                                               std::hash<Key>,
+                                               std::equal_to<Key>,
+                                               Eigen::aligned_allocator<std::pair<const Key, Value>>>;
+
+using VectorIsometry3d = AlignedVector<Eigen::Isometry3d>;
+using VectorVector4d = AlignedVector<Eigen::Vector4d>;
+using VectorVector3d = std::vector<Eigen::Vector3d>;
+using TransformMap = AlignedMap<std::string, Eigen::Isometry3d>;
+
+typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> TrajArray;
```

Comment:
This also seems a bit odd that it is in environment.

Reply:
I think it may be good to break these common type and the macro stuff and put it into a tesseract_common package that all use. What do you think?

Reply:
That would make sense to me.